src/genophenocorr/model/_protein.py

```diff
@@ -242,6 +246,18 @@ def motifs(self) -> typing.Iterable[ProteinFeature]:
         """
         return filter(lambda f: f.feature_type == FeatureType.MOTIF, self.protein_features)
 
+    def get_features_variant_overlaps(self, var_start: int, var_end: int) -> typing.Sequence[ProteinFeature]:
+        affected_features = set()
+        for feat in self.protein_features:
+            if feat.info.start is None or feat.info.end is None:
+                print(f"{feat.info.name} has no start and end info")
+                continue
+            if feat.info.start <= var_start <= feat.info.end:
+                affected_features.add(feat.to_string())
```

Comment:
`ProteinFeature` has no `to_string` method but `SimpleProteinFeature` has it. This code will fail when `FeatureInfo` is provided with a different `ProteinFeature` implementation, which is certainly a legal thing to do.

This bug as well as the one below are glaring into my eyes in PyCharm thanks to the typing annotation and the PyCharm linter. Please consider setting up a Python linter in Code, or switching to PyCharm. :roll_eyes:

Reply:
It looks like MyPy linter can catch these bugs:

![image](https://github.com/monarch-initiative/genophenocorr/assets/12170955/c32bccc9-0155-4b59-8a53-9d4ef4ebd558)

![image](https://github.com/monarch-initiative/genophenocorr/assets/12170955/83dc8f00-80d4-4827-bc9d-1e1fbd0555f0)